### DISCUSSION TITLE
写了个简单路由，把开发机的Cookie发布到自建docker服务器后，无法登录了

### DISCUSSION THREAD

**Discussion by shawn8888** [DIYgod/RSSHub#8996] [Q&A] [answered]
自己写了一个简单的路由，用自己的帐号登录论坛，抓取更新的帖子，思路是：
1.	Puppeteer 方式关掉headless ，调出浏览器后用自己的帐号登录一个论坛，把cookie保存到json文件
2.	打开headless，读取保存好的cookie文件，可以正常登录。抓取更新。
在Windows调试机上运行通过，但是把路由和cookie文件copy到自建的docker服务器，运行出错了。因为用cookie登录失败。同样的代码和cookie，换了个puppeteer环境，就无法登录了。

问题：
1.	看来这个论坛保存的cookie可以识别浏览器。我比对了一下开发环境和docker环境的useragent (config.js -> ua: envs.UA)，一模一样啊。那么论坛服务器是怎么知道两个puppeteer不一样的呢？
2.	Docker中RSSHUB用的是 browserless/chrome的镜像。我想：把windows 开发机的puppeteer也用docker中的这个puppeteer不就解决了？
于是改了config.js中的puppeteerWSEndpoint: 'ws://mydocker.IP:3000',
但是docker中的puppeteer是无头的。我无法通过headless: false 打开浏览器手动登录。（因为登录有校验码，所以不方便无头方式登录）
还尝试了在浏览器中打开 dockerip:3000 ，合乎是个debug环境，不知道怎么在rsshub中使用它。
还望高人指点迷津。谢谢！

**Comment by shawn8888** [comment, marked as answer]
OK，解决了！
问题还是出在useragent上。不知道为什么，RSSHub没有设置 Puppeteer  的useragent
手动添加的方法是：

```js
            const page = await browser.newPage();
            await page.setUserAgent('Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.108 Safari/537.36');            

```

**Reply by TonyRL** [reply]
https://github.com/DIYgod/RSSHub/blob/97429ad0161a133087f31b3163b1ae306f425829/lib/utils/puppeteer.js#L5
UA is set here. Probably it's too old.
